### ISSUE TITLE
rc6 language custom makes pibox-installer crashs

### ISSUE THREAD

**Issue by rgaudin** [kiwix/kiwix-hotspot#133]
In GitLab by @Kelson42 on Oct 21, 2017, 05:06

Have configured french and....

```
END_OF_CMD3267'
sudo chown ideascube:ideascube /opt/venvs/ideascube/lib/python3.4/site-packages/ideascube/conf/pibox.py
sudo sh -c 'cat > /opt/venvs/ideascube/lib/python3.4/site-packages/ideascube/conf/kb.py <<END_OF_CMD3267
from .pibox import *  # pragma: no flakes

EXTRA_APP_CARDS = ['\''khanacademy'\'']

CUSTOM_CARDS = [{'\''url'\'': '\''http://aflatoun.koombook.lan'\'', '\''category'\'': '\''learn'\'', '\''is_staff'\'': False, '\''fa'\'': '\''book'\'', '\''title'\'': '\''Aflatoun'\'', '\''description'\'': '\''Social and Financial Education for Children and Young People'\''}, {'\''url'\'': '\''http://fr.wikifundi.koombook.lan'\'', '\''category'\'': '\''create'\'', '\''is_staff'\'': False, '\''fa'\'': '\''wikipedia-w'\'', '\''title'\'': '\''Wikifundi'\'', '\''description'\'': '\''Environnement qui vous permet de créer des articles Wikipédia hors-ligne'\''}, {'\''url'\'': '\''http://edupi.koombook.lan'\'', '\''category'\'': '\''manage'\'', '\''is_staff'\'': False, '\''fa'\'': '\''folder'\'', '\''title'\'': '\''Edupi'\'', '\''description'\'': '\''Content management application'\''}]

LANGUAGE_CODE = '\''fr'\''

LANGUAGES = [('\''fr'\'', '\''Français'\'')]]

END_OF_CMD3267'
sudo chown ideascube:ideascube /opt/venvs/ideascube/lib/python3.4/site-packages/ideascube/conf/kb.py
create super user
[36mIDEASCUBE_ID=kb[0m
STDERR: Traceback (most recent call last):
STDERR:   File "/opt/venvs/ideascube/bin/django-admin", line 11, in <module>
STDERR:     load_entry_point('Django==1.11.1', 'console_scripts', 'django-admin')()
STDERR:   File "/opt/venvs/ideascube/lib/python3.4/site-packages/django/core/management/__init__.py", line 363, in execute_from_command_line
STDERR:     utility.execute()
STDERR:   File "/opt/venvs/ideascube/lib/python3.4/site-packages/django/core/management/__init__.py", line 307, in execute
STDERR:     settings.INSTALLED_APPS
STDERR:   File "/opt/venvs/ideascube/lib/python3.4/site-packages/django/conf/__init__.py", line 56, in __getattr__
STDERR:     self._setup(name)
STDERR:   File "/opt/venvs/ideascube/lib/python3.4/site-packages/django/conf/__init__.py", line 41, in _setup
STDERR:     self._wrapped = Settings(settings_module)
STDERR:   File "/opt/venvs/ideascube/lib/python3.4/site-packages/django/conf/__init__.py", line 110, in __init__
STDERR:     mod = importlib.import_module(self.SETTINGS_MODULE)
STDERR:   File "/opt/venvs/ideascube/lib/python3.4/importlib/__init__.py", line 109, in import_module
STDERR:     return _bootstrap._gcd_import(name[level:], package, level)
STDERR:   File "<frozen importlib._bootstrap>", line 2254, in _gcd_import
STDERR:   File "<frozen importlib._bootstrap>", line 2237, in _find_and_load
STDERR:   File "<frozen importlib._bootstrap>", line 2226, in _find_and_load_unlocked
STDERR:   File "<frozen importlib._bootstrap>", line 1200, in _load_unlocked
STDERR:   File "<frozen importlib._bootstrap>", line 1129, in _exec
STDERR:   File "<frozen importlib._bootstrap>", line 1471, in exec_module
STDERR:   File "<frozen importlib._bootstrap>", line 321, in _call_with_frames_removed
STDERR:   File "/opt/venvs/ideascube/lib/python3.4/site-packages/ideascube/settings.py", line 26, in <module>
STDERR:     sub = importlib.import_module(".conf." + IDEASCUBE_ID, package="ideascube")
STDERR:   File "/opt/venvs/ideascube/lib/python3.4/importlib/__init__.py", line 109, in import_module
STDERR:     return _bootstrap._gcd_import(name[level:], package, level)
STDERR:   File "<frozen importlib._bootstrap>", line 2254, in _gcd_import
STDERR:   File "<frozen importlib._bootstrap>", line 2237, in _find_and_load
STDERR:   File "<frozen importlib._bootstrap>", line 2226, in _find_and_load_unlocked
STDERR:   File "<frozen importlib._bootstrap>", line 1200, in _load_unlocked
STDERR:   File "<frozen importlib._bootstrap>", line 1129, in _exec
STDERR:   File "<frozen importlib._bootstrap>", line 1467, in exec_module
STDERR:   File "<frozen importlib._bootstrap>", line 1572, in get_code
STDERR:   File "<frozen importlib._bootstrap>", line 1532, in source_to_code
STDERR:   File "<frozen importlib._bootstrap>", line 321, in _call_with_frames_removed
STDERR:   File "/opt/venvs/ideascube/lib/python3.4/site-packages/ideascube/conf/kb.py", line 9
STDERR:     LANGUAGES = [('fr', 'Français')]]
STDERR:                                     ^
STDERR: SyntaxError: invalid syntax
sudo shutdown 0
STDERR: Shutdown scheduled for Fri 2017-10-20 22:54:04 CEST, use 'shutdown -c' to cancel.
Failed
ssh command failed with status 1. cmd: sudo ideascube createsuperuser --serial 'admin' <<< 'admin'
```

**Comment by rgaudin** [comment]
In GitLab by @thiolliere on Oct 22, 2017, 16:38

Kb.py is wrongly written by pibox installer. I think it comes from @mgautierfr changes. I can't really debug as I'm busy traveling around Europe and so I'll be for ~2 weeks

**Comment by rgaudin** [comment]
In GitLab by @Kelson42 on Oct 22, 2017, 16:40

@thiolliere Thx. Enjoy your trip. Let me know if you are in Zurich!
@mgautierfr Could you please have a look quickly (and release rc7)? It almost looks like a typo to me.

**Comment by rgaudin** [comment]
In GitLab by @mgautierfr on Oct 23, 2017, 12:11

mentioned in commit 0c33a9861bde7f68fa3e818661b53d5e80c779ea

**Comment by rgaudin** [comment]
In GitLab by @mgautierfr on Oct 23, 2017, 12:15

mentioned in merge request !17

**Comment by rgaudin** [comment]
In GitLab by @mgautierfr on Oct 23, 2017, 12:16

mentioned in commit 25c9aff150e5cba8a89ffc8a0a5e4bbc280c6391

**Comment by rgaudin** [comment]
In GitLab by @mgautierfr on Oct 23, 2017, 12:17

closed via commit 5437602ee619ff7a6f725a86fe0c503c2418ef67

**Comment by rgaudin** [comment]
In GitLab by @mgautierfr on Oct 23, 2017, 12:17

closed via commit 25c9aff150e5cba8a89ffc8a0a5e4bbc280c6391
